src/server/frontend_wayland/ext_image_capture_v1.cpp

```diff
@@ -0,0 +1,604 @@
+    wl_resource* resource)
+    : wayland::OutputImageCaptureSourceManagerV1{resource, Version<1>()}
+{
+}
+
+void mf::ExtOutputImageCaptureSourceManagerV1::create_source(wl_resource* new_resource, wl_resource* output)
+{
+    auto& output_global = OutputGlobal::from_or_throw(output);
+    new ExtImageCaptureSourceV1{new_resource, output_global, this};
+}
+
+mf::ExtImageCaptureSourceV1::ExtImageCaptureSourceV1(
+    wl_resource* resource, OutputGlobal& output,
+    ExtOutputImageCaptureSourceManagerV1 *manager)
+    : wayland::ImageCaptureSourceV1{resource, Version<1>()},
+      output{&output},
+      manager{manager}
+{
+}
+
+mf::ExtImageCaptureSourceV1* mf::ExtImageCaptureSourceV1::from_or_throw(wl_resource *resource)
+{
+    auto source = dynamic_cast<ExtImageCaptureSourceV1*>(wayland::ImageCaptureSourceV1::from(resource));
+
+    if (!source)
+    {
+        BOOST_THROW_EXCEPTION(std::runtime_error(
+            "ext_image_capture_source_v1@" +
+            std::to_string(wl_resource_get_id(resource)) +
+            " is not a mir::frontend::ExtImageCaptureSourceV1"));
+    }
+    return source;
+}
+
+auto mf::create_ext_image_copy_capture_manager_v1(
+    wl_display* display,
+    std::shared_ptr<Executor> const& wayland_executor,
+    std::shared_ptr<graphics::GraphicBufferAllocator> const& allocator,
+    std::shared_ptr<compositor::ScreenShooterFactory> const& screen_shooter_factory,
+    std::shared_ptr<SurfaceStack> const& surface_stack)
+-> std::shared_ptr<wayland::ImageCopyCaptureManagerV1::Global>
+{
+    auto ctx = std::make_shared<ExtImageCaptureV1Ctx>(
+        wayland_executor, allocator, screen_shooter_factory, surface_stack);
+    return std::make_shared<ExtImageCopyCaptureManagerV1Global>(display, std::move(ctx));;
+}
+
+mf::ExtImageCopyCaptureManagerV1Global::ExtImageCopyCaptureManagerV1Global(
+    wl_display *display,
+    std::shared_ptr<ExtImageCaptureV1Ctx> const& ctx)
+    : Global{display, Version<1>()},
+      ctx{ctx}
+{
+}
+
+void mf::ExtImageCopyCaptureManagerV1Global::bind(wl_resource *new_resource)
+{
+    new ExtImageCopyCaptureManagerV1{new_resource, ctx};
+}
+
+mf::ExtImageCopyCaptureManagerV1::ExtImageCopyCaptureManagerV1(
+    wl_resource *resource,
+    std::shared_ptr<ExtImageCaptureV1Ctx> const& ctx)
+    : wayland::ImageCopyCaptureManagerV1{resource, Version<1>()},
+      ctx{ctx}
+{
+}
+
+void mf::ExtImageCopyCaptureManagerV1::create_session(
+    wl_resource* new_resource, wl_resource* source, uint32_t options)
+{
+    auto const source_instance = ExtImageCaptureSourceV1::from_or_throw(source);
+    new ExtImageCopyCaptureSessionV1{new_resource, source_instance, options, ctx};
+}
+
+void mf::ExtImageCopyCaptureManagerV1::create_pointer_cursor_session(
+    [[maybe_unused]] wl_resource* new_resource,
+    [[maybe_unused]] wl_resource* source,
+    [[maybe_unused]] wl_resource* pointer)
+{
+    BOOST_THROW_EXCEPTION(std::runtime_error("not implemented"));
```

Comment:
Not sure what the right thing to do here is. But `wayland::ProtocolError()` seems more likely to be a graceful fail. Or a `fatal_error()`

Reply:
This will be translated into an [`implementation`](https://wayland.app/protocols/wayland#wl_display:enum:error:entry:implementation) protocol error by our Wayland machinery; this is a reasonable approach.

Reply:
None of the `ext_image_copy_capture_manager_v1::error` values really match up to a missing implementation. So I think this is probably okay as is for now. It'll be properly resolved in a followup PR.